presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java

```diff
@@ -651,15 +712,32 @@ private static void assertFileContentsPresto(
             Format format,
             boolean isHiveWriter,
             boolean useSelectiveOrcReader,
-            List<Map<Integer, Map<Subfield, TupleDomainFilter>>> filters)
-            throws IOException
+            List<Map<Integer, Map<Subfield, TupleDomainFilter>>> filters,
+            Optional<List<List<Integer>>> desiredOrders)
+            throws Exception
     {
         OrcPredicate orcPredicate = createOrcPredicate(types, expectedValues, format, isHiveWriter);
         if (useSelectiveOrcReader) {
             assertFileContentsPresto(types, tempFile.getFile(), expectedValues, orcEncoding, orcPredicate, Optional.empty(), ImmutableList.of(), ImmutableMap.of(), ImmutableMap.of());
+            for (int i = 0; i < filters.size(); i++) {
+                Map<Integer, Map<Subfield, TupleDomainFilter>> columnFilters = filters.get(i);
+                List<List<?>> filteredRows = filterRows(types, expectedValues, columnFilters);
+
+                // Turn normal TupleDomainFilters into their corresponding OrderTrackingOrcFilters.
+                // Keeping types the same so that we can reuse the "filters" variable;
+                Optional<TupleDomainFilterOrderChecker> orderChecker = Optional.empty();
```

Comment:
this can be simplified like so:

```
                Optional<TupleDomainFilterOrderChecker> orderChecker = Optional.empty();
                if(!expectedFilterOrder.isEmpty()) {
                    orderChecker = Optional.of(new TupleDomainFilterOrderChecker(expectedFilterOrder.get(i)));
                }

                Optional<Map<Integer, Map<Subfield, TupleDomainFilter>>> transformedFilters = Optional.of(orderChecker.map(checker -> addOrderTracking(columnFilters, checker)).orElse(columnFilters));

                assertFileContentsPresto(types, tempFile.getFile(), filteredRows , orcEncoding, orcPredicate, transformedFilters,  ImmutableList.of(), ImmutableMap.of(), ImmutableMap.of());

                orderChecker.ifPresent(TupleDomainFilterOrderChecker::assertOrder);




    private static Map<Integer, Map<Subfield, TupleDomainFilter>> addOrderTracking(Map<Integer, Map<Subfield, TupleDomainFilter>> filters, TupleDomainFilterOrderChecker orderChecker)
    {
        return Maps.transformEntries(filters, (column, columnFilters) -> Maps.transformValues(columnFilters, filter -> addOrderTracking(filter, orderChecker, column)));
    }
```

Reply:
nice